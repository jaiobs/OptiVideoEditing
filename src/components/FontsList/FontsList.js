import React, { Component } from "react";
import {
  Text,
  TouchableOpacity,
  FlatList,
  ScrollView,
} from "react-native";

export default class FontsList extends Component {
  constructor(props) {
    super(props)
      this.state = {
      }
    }

    render() {
        const {fontsList,isSelected} = this.props
        return(
          <ScrollView keyboardShouldPersistTaps={'always'} keyboardDismissMode='on-drag' style={{flex: 1, zIndex:999}}>
            <FlatList
            data={fontsList}
            keyboardShouldPersistTaps={'always'}
            horizontal={true}
            renderItem={({ item }) => (
              <TouchableOpacity
                style={{height: 35, padding:3,marginTop:8, justifyContent:"center", flex:1, minWidth: 40, borderColor:"#fff", borderRadius:5,borderWidth: isSelected == item.value ? 0.7 : 0 , backgroundColor:"transparent", margin:2}}
                onPress={() => {this.props.onSelectFont(item)}}
              >
                  <Text style={{textAlign:"center",color:"#fff", marginTop: -3,fontFamily:item.value, fontSize: 18}}>{item.title}</Text>
                  </TouchableOpacity>
            )}
            keyExtractor={item => item.id}
            />
            </ScrollView>
        )
    }
}