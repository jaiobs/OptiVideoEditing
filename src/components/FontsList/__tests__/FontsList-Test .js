import React from 'react';
import renderer from 'react-test-renderer';
import FontsList from '../FontsList';

jest.useFakeTimers();

const mockNullData = [];
const mockData = [
  {
    title: 'Verdana',
    value: 'Verdana-Bold',
  },
];

describe('FontsList Tests', () => {
  it('should render FontsList with data correctly', async () => {
    const tree = renderer.create(<FontsList data={mockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render FontsList with null data', async () => {
    const tree = renderer.create(<FontsList data={mockNullData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
