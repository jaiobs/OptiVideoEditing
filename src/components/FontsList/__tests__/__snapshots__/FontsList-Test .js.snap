// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontsList Tests should render FontsList with data correctly 1`] = `
<RCTScrollView
  keyboardDismissMode="on-drag"
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "flex": 1,
      "zIndex": 999,
    }
  }
>
  <View>
    <RCTScrollView
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      keyboardShouldPersistTaps="always"
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTScrollView>
`;

exports[`FontsList Tests should render FontsList with null data 1`] = `
<RCTScrollView
  keyboardDismissMode="on-drag"
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "flex": 1,
      "zIndex": 999,
    }
  }
>
  <View>
    <RCTScrollView
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      keyboardShouldPersistTaps="always"
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTScrollView>
`;
